topics/single-cell/tutorials/scrna-data-ingest/tutorial.md

```diff
@@ -0,0 +1,881 @@
+>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
+>
+{: .hands_on}
+
+{% snippet faqs/galaxy/histories_copy_dataset.md %}
+
+First, we will extract observations and the full matrix from our AnnData.
+
+> <tip-title>Skip some steps!</tip-title>
+>
+> If you are following this entire tutorial (rather than using a specific section necessary!), you can stay in the your previous history and just reuse outputs to build different single cell objects!
+{: .tip}
+
+> <hands-on-title> Inspect AnnData </hands-on-title>
+>
+> 1. {% tool [Inspect AnnData](toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.3+galaxy0) %} with the following parameters:
+>    - {% icon param-file %} *"Annotated data matrix"*: `Downsampled AnnData object`
+>    - *"What to inspect?"*: `Key-indexed observations annotation (obs)`
+>   
+> 2. **Rename** {% icon galaxy-pencil %} the output `Observations`.
+>
+> 3. {% tool [Inspect AnnData](toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.3+galaxy0) %} with the following parameters:
+>    - {% icon param-file %} *"Annotated data matrix"*: `Downsampled AnnData object`
+>    - *"What to inspect?"*: `The full data matrix`
+>
+> 4. **Rename** {% icon galaxy-pencil %} the output `Matrix`.
+>    
+{: .hands_on}
+
+> <question-title></question-title>
+>
+> What are the rows, and what are the columns in the retrieved Matrix?
+>
+> > <solution-title></solution-title>
+> >
+> > If you just click on the `Matrix` dataset, you will see a preview, showing barcodes in the first column, while genes are in the first row.
+> > 
+> {: .solution}
+>
+{: .question}
+
+However, the next tool we need expects a matrix wherein the genes are listed in the first column and the barcodes are listed in the first row. Therefore, we need to transpose the current matrix.
+
+> <hands-on-title> Transpose the matrix </hands-on-title>
+>
+> 1. {% tool [Transpose](toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.8+galaxy0) %} with the following parameters:
+>    - {% icon param-file %} *"Input tabular dataset"*: `Matrix`
+>
+{: .hands_on}
+
+And now we are ready to input that data to **DropletUtils** tool.
+
+> <hands-on-title> DropletUtils </hands-on-title>
+>
+> 1. {% tool [DropletUtils](toolshed.g2.bx.psu.edu/repos/iuc/dropletutils/dropletutils/1.10.0+galaxy2) %} with the following parameters:
+>    - *"Format for the input matrix"*: `Tabular`
+>    - {% icon param-file %} *"Count Data"*: output of **Transpose** {% icon tool %}
+>    - *"Operation"*: `Filter for Barcodes`
+>        - *"Method"*: `DefaultDrops`
+>            - *"Expected Number of Cells"*: `338`
+>            - *"Upper Quantile"*: `1.0`
+>            - *"Lower Proportion"*: `0.0`
+>        - *"Format for output matrices"*: `Bundled (barcodes.tsv, genes.tsv, matrix.mtx)`
+>        - *"Random Seed"*: `100`
+>
+{: .hands_on}
+
+Finally, let's combine those files that we have just generated and turn them into the SingleCellExperiment!
```

Comment:
Alternatively, can be done with SCEasy. Anndata -> Seurat -> SCE

Reply:
Right, will add an additional hands-on box!